### ISSUE TITLE
Constraint Based Direct Fourier Method does not return a child dataset

### ISSUE THREAD

**Issue by ercius** [OpenChemistry/tomviz#1301]
The referenced recon method does not create a Reconstruction child data set. It instead updates the original data:
```
        # Set the result as the new scalars.
        utils.set_array(dataset, recon)

        # Mark dataset as volume
        utils.mark_as_volume(dataset)
```
The Weighted Backprojection does return as Reconstruction child dataset:
```
        # Set up the output dataset
        from vtk import vtkImageData
        recon_dataset = vtkImageData()
        recon_dataset.CopyStructure(dataset)
        utils.set_array(recon_dataset, recon)
        utils.mark_as_volume(recon_dataset)

        returnValues = {}
        returnValues["reconstruction"] = recon_dataset
        return returnValues
```

**Comment by Hovden** [comment]
Good catch @ercius . This is also the case for the Constraint-based Direct Fourier
@yijiang1 may not be aware of this updated feature.

**Comment by mathturtle** [comment]
So this is actually the legacy of a UI dispute between Yi and me. :(  If a python operator returns a child dataset, it has to use the new JSON specification for its parameters.  Yi disliked the JSON-generated UI for this reconstruction and wanted to keep the original custom UI that he wrote... so I didn't update this one when I changed all the reconstructions to produce child datasets.  See #764 and the associated PR for some of the discussion (some was at meetings I think, because I remember more than is in the issue).

I can't find the commit I had to fix this last reconstruction operator though.  So I'll have to redo it.

**Comment by cryos** [comment]
Let's track this in this issue, and make sure we come to a conclusion on producing child data sets. If there are blockers we want to get them into issues, and we can decide how/when we want to address them. Thanks for bringing this up, I hadn't gone through all the reconstructions but had thought they were all producing child data sets now.

**Comment by mathturtle** [comment]
After a bit of digging through git's more obscure commands, I found my work-in-progress commit for this operator.  I'm cherry-picking it now and I'll push it as a branch in a few minutes.

**Comment by cryos** [comment]
This has been resolved, and was part of the 1.2.0 release.
